rdflib/plugins/stores/sparqlstore.py

```diff
@@ -145,6 +109,84 @@ def CastToTerm(node):
         raise Exception('Unknown answer type')
 
 
+def CastToTerm(node):
+    warnings.warn(
+        "Call to deprecated function CastToTerm, use _node_from_result.",
+        category=DeprecationWarning,
+    )
+    return _node_from_result(node)
+
+
+def _node_to_sparql(node):
+    if isinstance(node, BNode):
+        raise Exception(
+            "SPARQLStore does not support BNodes! "
```

Comment:
i thought so too, but as this currently raises an overly broad exception, the only thing people can do is rely on its message to actually handle it. That's why I decided against it for now and didn't change the message.

maybe we should have an issue to change all `raise Exception` to something more appropriate in 5.0.0